### ISSUE TITLE
EnderCompass v2.0 - MC 1.15.2 - FTB Modpack

### ISSUE THREAD

**Issue by Drjoachim** [Tfarcenim/EnderCompass#4]
Rightclicking it gives this error in the logs.
Also it does not point to anything, just spinning around.

`[31May2020 18:52:09.373] [Server thread/FATAL] [net.minecraftforge.fml.loading.RuntimeDistCleaner/DISTXFORM]: Attempted to load class net/minecraft/client/Minecraft for invalid dist DEDICATED_SERVER
[31May2020 18:52:09.373] [Server thread/FATAL] [net.minecraft.util.concurrent.ThreadTaskExecutor/]: Error executing task on Server
java.lang.RuntimeException: Attempted to load class net/minecraft/client/Minecraft for invalid dist DEDICATED_SERVER
        at net.minecraftforge.fml.loading.RuntimeDistCleaner.processClass(RuntimeDistCleaner.java:71) ~[forge-1.15.2-31.1.87.jar:31.1]
        at cpw.mods.modlauncher.serviceapi.ILaunchPluginService.processClass(ILaunchPluginService.java:103) ~[modlauncher-5.1.0.jar:5.1.0+69+master.79f13f7]
        at cpw.mods.modlauncher.LaunchPluginHandler.offerClassNodeToPlugins(LaunchPluginHandler.java:85) ~[modlauncher-5.1.0.jar:?]
        at cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:115) ~[modlauncher-5.1.0.jar:?]
        at cpw.mods.modlauncher.TransformingClassLoader$DelegatedClassLoader.findClass(TransformingClassLoader.java:239) ~[modlauncher-5.1.0.jar:?]
        at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:126) ~[modlauncher-5.1.0.jar:?]
        at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:96) ~[modlauncher-5.1.0.jar:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]
        at net.minecraftforge.fml.network.simple.SimpleChannel.sendToServer(SimpleChannel.java:104) ~[?:?]
        at io.github.mribby.endercompass.network.CPacketGetStrongholdPos.lambda$handle$0(CPacketGetStrongholdPos.java:27) ~[?:2.0]
        at net.minecraftforge.fml.network.NetworkEvent$Context.enqueueWork(NetworkEvent.java:215) ~[?:?]
        at io.github.mribby.endercompass.network.CPacketGetStrongholdPos.handle(CPacketGetStrongholdPos.java:24) ~[?:2.0]
        at net.minecraftforge.fml.network.simple.IndexedMessageCodec.lambda$tryDecode$3(IndexedMessageCodec.java:128) ~[?:?]
        at java.util.Optional.ifPresent(Optional.java:183) ~[?:?]
        at net.minecraftforge.fml.network.simple.IndexedMessageCodec.tryDecode(IndexedMessageCodec.java:128) ~[?:?]
        at net.minecraftforge.fml.network.simple.IndexedMessageCodec.consume(IndexedMessageCodec.java:162) ~[?:?]
        at net.minecraftforge.fml.network.simple.SimpleChannel.networkEventListener(SimpleChannel.java:80) ~[?:?]
        at net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:212) ~[eventbus-2.2.0-service.jar:?]
        at net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:204) ~[eventbus-2.2.0-service.jar:?]
        at net.minecraftforge.eventbus.EventBus.post(EventBus.java:258) ~[eventbus-2.2.0-service.jar:?]
        at net.minecraftforge.fml.network.NetworkInstance.dispatch(NetworkInstance.java:84) ~[?:?]
        at net.minecraftforge.fml.network.NetworkHooks.lambda$onCustomPayload$1(NetworkHooks.java:78) ~[?:?]
        at java.util.Optional.map(Optional.java:265) ~[?:?]
        at net.minecraftforge.fml.network.NetworkHooks.onCustomPayload(NetworkHooks.java:78) ~[?:?]
        at net.minecraft.network.play.ServerPlayNetHandler.func_147349_a(ServerPlayNetHandler.java:1279) ~[?:?]
        at net.minecraft.network.play.client.CCustomPayloadPacket.func_148833_a(CCustomPayloadPacket.java:42) ~[?:?]
        at net.minecraft.network.play.client.CCustomPayloadPacket.func_148833_a(CCustomPayloadPacket.java:12) ~[?:?]
        at net.minecraft.network.PacketThreadUtil.func_225383_a(SourceFile:21) ~[?:?]
        at net.minecraft.util.concurrent.TickDelayedTask.run(SourceFile:18) ~[?:?]
        at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213166_h(SourceFile:144) [?:?]
        at net.minecraft.util.concurrent.RecursiveEventLoop.func_213166_h(SourceFile:23) [?:?]
        at net.minecraft.server.MinecraftServer.func_213166_h(MinecraftServer.java:731) [?:?]
        at net.minecraft.server.MinecraftServer.func_213166_h(MinecraftServer.java:141) [?:?]
        at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213168_p(SourceFile:118) [?:?]
        at net.minecraft.server.MinecraftServer.func_213205_aW(MinecraftServer.java:714) [?:?]
        at net.minecraft.server.MinecraftServer.func_213168_p(MinecraftServer.java:708) [?:?]
        at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213161_c(SourceFile:127) [?:?]
        at net.minecraft.server.MinecraftServer.func_213202_o(MinecraftServer.java:694) [?:?]
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:641) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
`

**Comment by Charmanth** [comment]
I'm getting the same issue on a dedicated minecraft server.  I really liked this mod for 1.12.2 so I'm kind of bummed it doesn't work for 1.15.2

**Comment by Charmanth** [comment]
So I did a test with just endercompass 2.0: It works by itself in single player, but my server has something like 101 modfiles, so it has to be a mod conflict.  

EDIT: Possible issue with repurposed structures?  By default it edits the stronghold, but I don't know if it would edit the end portal itself.

Edit: For me at least I found that the issue is on servers and not just client-side/single player.  My pack works fine in single player.
